code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -132,7 +132,7 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	med_hud_set_health()
 	med_hud_set_status()
 	if(!QDELETED(summoner))
-		if(summoner.stat == DEAD)
+		if(summoner.stat == DEAD || HAS_TRAIT(summoner, TRAIT_NODEATH) && summoner.health <= HEALTH_THRESHOLD_DEAD)
```

Comment:
no reason to check for the trait, just do the health threshold check and leave a comment for why it's there